tests/test_test_functions.py

```diff
@@ -97,3 +97,50 @@ def test_total_sensitivity_index():
                          0.222116249, 0.030859879, 0.032170899])
 
     assert_allclose(actual, expected, atol=1e-2, rtol=1e-6)
+
+
+def test_Modified_Sobol_G():
+    parameter_values = np.zeros((1, 8))
+    delta_values = np.ones_like(parameter_values)
+    alpha_values = np.array([2]*8)
+    actual = evaluate(parameter_values, delta=delta_values, alpha=alpha_values)
+    expected = np.array([10.6275])
+    assert_allclose(actual, expected, atol=1e-4, rtol=1e-4)
+
+
+def test_Modified_Sobol_G_error_if_type_wrong():
+
+    parameter_values = np.zeros((1, 8))
+    delta_values = [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8]
+    with raises(TypeError):
+        evaluate(parameter_values, delta=delta_values)
+
+    alpha_values = [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8]
+    with raises(TypeError):
+        evaluate(parameter_values, alpha=alpha_values)
+
+
+def test_Modified_Sobol_G_error_if_value_beyond_range():
```

Comment:
It's worth testing the error messages too. Something like:
```python
with raises(ValueError) as ex:
    <function call which raises the error>
assert "The error message I expect" in str(ex)
```

Reply:
Hi @QianWanghhu and @willu47 

Note that with later versions of pytest the `in str(ex)` approach will error. The alternative is either:

```python
with raises(ValueError) as ex:
    <function call which raises the error>

assert "The error message I expect" in str(ex.value)

# OR

# `match` accepts regex if desired
with pytest.raises(ValueError, match="<expected error message>"):
    <function call which raises the error>
```

I had to adjust this a few months ago, see:
https://github.com/SALib/SALib/commit/9fe3c129ae5476c38b5c2d7622565635dabc287f

Reply:
Hi @willu47 and @ConnectedSystems 
Thanks for your comments and suggestions which are very helpful. I will fix it accordingly.